<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InterfaceEditorWidget</class>
 <widget class="QWidget" name="InterfaceEditorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>560</width>
    <height>350</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::FieldGrowthPolicy::ExpandingFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="nameLabel">
       <property name="text">
        <string>Name:</string>
       </property>
       <property name="toolTip">
        <string>Name of the physical interface (e.g. eth0, ens192, bond0). This must match the actual interface name on the host.</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="ifaceName">
       <property name="toolTip">
        <string>Name of the physical interface (e.g. eth0, ens192, bond0). This must match the actual interface name on the host.</string>
       </property>
       <property name="placeholderText">
        <string>e.g. eth0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelLabel">
       <property name="text">
        <string>Label:</string>
       </property>
       <property name="toolTip">
        <string>Optional user-friendly label describing the role of this interface (e.g. outside, inside, dmz).</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="ifaceLabel">
       <property name="toolTip">
        <string>Optional user-friendly label describing the role of this interface (e.g. outside, inside, dmz).</string>
       </property>
       <property name="placeholderText">
        <string>e.g. outside</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="typeLabel">
       <property name="text">
        <string>Type:</string>
       </property>
       <property name="toolTip">
        <string>Select how this interface obtains its IP address. Static: configure addresses below. Dynamic: gets address via DHCP or PPP. Unnumbered: no IP address (e.g. tunnel endpoint).</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="ifaceType">
       <property name="toolTip">
        <string>Select how this interface obtains its IP address. Static: configure addresses below. Dynamic: gets address via DHCP or PPP. Unnumbered: no IP address (e.g. tunnel endpoint).</string>
       </property>
       <item>
        <property name="text">
         <string>Static IP address</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Dynamic IP address (DHCP/PPP)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Unnumbered interface</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="addressTable">
     <property name="toolTip">
      <string>IP addresses assigned to this interface. Add rows with the &quot;Add Address&quot; button below.</string>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
     </property>
     <property name="columnCount">
      <number>3</number>
     </property>
     <column>
      <property name="text">
       <string>IP Address</string>
      </property>
      <property name="toolTip">
       <string>IP address (e.g. 192.168.1.1 or 2001:db8::1)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Netmask</string>
      </property>
      <property name="toolTip">
       <string>Netmask in CIDR notation (e.g. 24) or dotted notation (e.g. 255.255.255.0)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Protocol</string>
      </property>
      <property name="toolTip">
       <string>Address family: IPv4 or IPv6</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="addressButtonLayout">
     <item>
      <widget class="QPushButton" name="addAddressButton">
       <property name="text">
        <string>Add Address</string>
       </property>
       <property name="toolTip">
        <string>Add a new IP address row for this interface</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeAddressButton">
       <property name="text">
        <string>Remove Address</string>
       </property>
       <property name="toolTip">
        <string>Remove the selected IP address row</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="addressSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="commentLabel">
     <property name="text">
      <string>Comment:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="ifaceComment">
     <property name="toolTip">
      <string>Optional comment for this interface</string>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>60</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
